src/createAnimatedComponent.tsx

```diff
@@ -639,13 +641,13 @@ export default function createAnimatedComponent(
           }
           if (sharedTransitionTag) {
             const sharedElementTransition =
-              this.props.sharedTransitionStyle ?? DefaultSharedTransition;
-            configureLayoutAnimations(
+              this.props.sharedTransitionStyle?.getInstance() ??
+              new SharedTransition();
```

Comment:
wouldn't it be better not to create a new empty transition every time here? Maybe just have some static field with default empty transition in SharedTransition class?

Reply:
Resolved in: https://github.com/software-mansion/react-native-reanimated/pull/4421/commits/5b95cfa3a40669568b028d97d8051adea2b261f0